src/TestUtils/src/UITest.Appium/HelperExtensions.cs

```diff
@@ -1660,23 +1660,53 @@ public static IList<object> GetPerformanceData(this IApp app, string performance
 		/// <summary>
 		/// Navigates back in the application by simulating a tap on the platform-specific back navigation button.
 		/// </summary>
-		/// <param name="app">Represents the main gateway to interact with an app.</param>
-		public static void TapBackArrow(this IApp app)
-		{
-			if (app is AppiumAndroidApp)
-			{
-				app.Tap(AppiumQuery.ByXPath("//android.widget.ImageButton[@content-desc='Navigate up']"));
-			}
-			else if (app is AppiumIOSApp || app is AppiumCatalystApp)
-			{
-				app.Tap(AppiumQuery.ByAccessibilityId("Back"));
-			}
-			else if (app is AppiumWindowsApp)
-			{
-				app.Tap(AppiumQuery.ByAccessibilityId("NavigationViewBackButton"));
+		/// <param name="app">The IApp instance representing the main gateway to interact with the application.</param>
+		/// <param name="customBackButtonIdentifier">Optional. The custom identifier for the back button. If not provided, default platform-specific identifiers will be used.</param>
+		public static void TapBackArrow(this IApp app, string customBackButtonIdentifier = "")
```

Comment:
Why not split in two methods?

```
static void TapBackArrow(this IApp app);
static void TapBackArrow(this IApp app, IQuery query);
```

The first one will just use the default platform-specific identifiers and the second one will allow any custom query like for example `AppiumQuery.ByAccessibilityId("Back")`